Add a transform option

searls opened the pull request:
Sometimes I want to stringify an object that might contain nested 
properties that need to be stringified specially. A `transform` 
function can be used to determine these special cases and return an
alternative string representation.

SBoudrias commented:
Interesting! I feel that's might be a generic way to address things like #39 (cc @stheine)

sindresorhus commented on readme.md:
```diff
@@ -79,6 +79,17 @@ Type: `Function`
 
+##### transform(obj, prop, originalResult)
+
+Type: `Function`
```
Needs a `<br>` at the end.

sindresorhus commented on readme.md:
```diff
@@ -79,6 +79,17 @@ Type: `Function`
+Type: `Function`
+Default: `undefined`
+
+Expected to return a `String` that transforms the string that resulted from
```
Don't hard-wrap the text.

sindresorhus replied:
`String` => `string`

sindresorhus commented on readme.md:
```diff
@@ -79,6 +79,17 @@ Type: `Function`
+stringifying `obj[prop]`. This can be used to detect special types of objects
+that need to be stringified in a particular way. The `transform` function might
+return an alternate string in this case, otherwise returning the
+`originalResult`.
```
Maybe add a link to the unit test to see how to use it. Not that clear from the description how it's supposed to be used.

sindresorhus commented on test.js:
```diff
@@ -72,6 +72,23 @@ it('considering filter option to stringify an object', () => {
 });
 
+it('allows an object to be transformed', () => {
+	const obj = {foo: {val: 10}, bar: 9, baz: [8]};
```
This would be more readable if formatted over multiple lines.

sindresorhus commented:
👍 Looks like a useful option. This could indeed solve many use-cases without adding more specific options.

searls commented:
hey, @sindresorhus -- I've addressed your feedback above. Looking forward to getting this merged & a release cut so I can move forward with releasing testdouble@2.  (This feature is a dependency of this feature for testdouble.js https://github.com/testdouble/testdouble.js/pull/203 )

sindresorhus commented:
Looks great. Thank you @searls :)

searls commented:
Thanks @sindresorhus! Appreciate your review, feedback, merge.
